lib/collection/proxy-config.js

```diff
@@ -118,13 +122,42 @@ _.assign(ProxyConfig.prototype, /** @lends ProxyConfig.prototype */ {
         }
 
         _.isString(options.match) && (this.match = new UrlMatchPattern(options.match));
+        _.isString(_.get(options, 'match.pattern')) && (this.match = new UrlMatchPattern(options.match.pattern));
         port && (this.port = port);
         _.isBoolean(options.tunnel) && (this.tunnel = options.tunnel);
         // todo: Add update method in parent class Property and call that here
         _.isBoolean(options.disabled) && (this.disabled = options.disabled);
     },
 
     /**
+     * Updates the protocols in the match pattern
+     *
+     * @param {Array.<String>} protocols The array of protocols
+     */
+    updateProtocols: function (protocols) {
+        if (!protocols) {
+            return;
+        }
+
+        var updatedProtocols,
+            updatedMatch,
+            hostAndPath = this.match.pattern.split(PROTOCOL_HOST_SEPARATOR)[1];
+
+        if (!hostAndPath) {
+            return;
+        }
+
+        _.isArray(protocols) && (updatedProtocols = _.intersection(AllOWED_PROTOCOLS, protocols));
```

Comment:
See https://lodash.com/docs/4.17.4#castArray, this will let you compute `updatedProtocols` in one go.